src/NosCore.GameObject/Providers/UpgradeService/UpgradeService.cs

```diff
@@ -0,0 +1,175 @@
+                    Language.Instance.GetMessageFromKey(LanguageKey.NOT_ENOUGH_MONEY, clientSession.Account.Language),
+                    SayColorType.Yellow));
+                return null;
+            }
+
+            if (clientSession.Character.Inventory.CountItem(1027) <
+                UpgradeHelper.Instance.SumSandAmount[item.ItemInstance.Upgrade + itemToSum.ItemInstance.Upgrade])
+            {
+                clientSession.SendPacket(clientSession.Character.GenerateSay(
+                    Language.Instance.GetMessageFromKey(LanguageKey.NOT_ENOUGH_ITEMS, clientSession.Account.Language),
+                    SayColorType.Yellow));
+                return null;
+            }
+
+            var random = (short)RandomFactory.Instance.RandomNumber();
+            if (random <=
+                UpgradeHelper.Instance.SumSuccessPercent[item.ItemInstance.Upgrade + itemToSum.ItemInstance.Upgrade])
+            {
+                HandleSuccessSum(clientSession, item, itemToSum);
+            }
+            else
+            {
+                HandleFailedSum(clientSession, item, itemToSum);
+            }
+
+            UpdateInv(clientSession, item, itemToSum);
+            clientSession.SendPacket(new ShopEndPacket
+            {
+                Type = ShopEndPacketType.CloseSubWindow
+            });
+
+            return item;
+        }
+
+        private void HandleSuccessSum(ClientSession clientSession, InventoryItemInstance item,
+            InventoryItemInstance itemToSum)
+        {
+            item.ItemInstance.Upgrade += (byte)(itemToSum.ItemInstance.Upgrade + 1);
+            item.ItemInstance.Item.DarkResistance += itemToSum.ItemInstance.Item.DarkResistance;
+            item.ItemInstance.Item.LightResistance += itemToSum.ItemInstance.Item.LightResistance;
+            item.ItemInstance.Item.FireResistance += itemToSum.ItemInstance.Item.FireResistance;
+            item.ItemInstance.Item.WaterResistance += itemToSum.ItemInstance.Item.WaterResistance;
+
+            clientSession.SendPacket(new PdtiPacket
+            {
+                Unknow = 10,
+                ItemVnum = item.ItemInstance.ItemVNum,
+                RecipeAmount = 1,
+                Unknow3 = 27,
+                ItemUpgrade = item.ItemInstance.Upgrade,
+                Unknow4 = 0
+            });
+            SendSumResult(clientSession, item, itemToSum, true);
+        }
+
+        private void HandleFailedSum(ClientSession clientSession, InventoryItemInstance item,
+            InventoryItemInstance itemToSum)
+        {
+            clientSession.Character.Inventory.RemoveItemAmountFromInventory(1, itemToSum.ItemInstanceId);
+            clientSession.Character.Inventory.RemoveItemAmountFromInventory(1, item.ItemInstanceId);
+            SendSumResult(clientSession, item, itemToSum, false);
+        }
+
+        private void SendSumResult(ClientSession clientSession, InventoryItemInstance item,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'item'.](https://app.codacy.com/app/NosCoreIO/NosCore/pullRequest?prid=4518428)